openedx/core/djangoapps/catalog/utils.py

```diff
@@ -101,18 +104,66 @@ def get_program_types(user=None):  # pylint: disable=redefined-builtin
         return []
 
 
-def get_programs_data(user=None):
-    """Return the list of Programs after adding the ProgramType Logo Image"""
+def _get_program_instructors(courses, program_id):
+    """
+    Returns the list of instructor from cached if cache key exists otherwise
+    iterate over the courses and return all the instructors of each course run
+    """
+    catalog_integration = CatalogIntegration.current()
+    if catalog_integration.enabled:
+        cache_key = '{base}.programs.instructors.{program_id}'.format(
+            base=catalog_integration.CACHE_KEY,
+            program_id=program_id
+        )
+        instructors = cache.get(cache_key) or []
+        if instructors:
+            return instructors
+
+        module_store = modulestore()
+        for key in _get_all_course_run_keys(courses):
+            descriptor = module_store.get_course(key)
+            if descriptor and descriptor.instructor_info:
+                for instructor in descriptor.instructor_info.get("instructors", []):
+                    instructors.append(instructor)
```

Comment:
We will need to account for duplicates here. Perhaps check the name of the instructor and don't append it there is an instructor already in the list with the same name.